include/jni/functions.hpp

```diff
@@ -589,7 +589,7 @@ namespace jni
     inline UniqueEnv AttachCurrentThread(JavaVM& vm)
        {
         JNIEnv* result;
-        CheckErrorCode(vm.AttachCurrentThread(&result, nullptr));
+        CheckErrorCode(vm.AttachCurrentThread(reinterpret_cast<void**>(&result), nullptr));
```

Comment:
@rayglover-ibm that's nice your fork has all the fixes I did and more maybe if you and @jfirebaugh agree that would be nicer to import your changes ?